src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -118,130 +86,74 @@ std::optional<std::array<double, Dim>> Translation<Dim>::inverse(
              << ").");
   double radial_function_value = 1.0;
   if (f_of_r_ != nullptr) {
-    std::array<double, Dim> distance_to_center = translated_coords;
-    double magnitude_function_of_time{0.};
-    double radius_squared = 0.;
+    std::array<double, Dim> distance_to_center = target_coords;
+    double squared_function_of_time{0.};
     double root_factor = 0.;
     for (size_t i = 0; i < Dim; i++) {
-      // shifted center and translated radius
+      // shifted center and target radius
       gsl::at(distance_to_center, i) -= gsl::at(center_, i);
-      radius_squared += square(gsl::at(distance_to_center, i));
       // squares of the function of time and root_factor needed for root
-      magnitude_function_of_time += square(function_of_time[i]);
+      squared_function_of_time += square(function_of_time[i]);
       root_factor +=
           -2.0 * gsl::at(distance_to_center, i) * function_of_time[i];
     }
-    const double translated_radius = sqrt(radius_squared);
-    const double& root =
-        root_finder(translated_radius, magnitude_function_of_time, root_factor);
+    double root = root_finder(magnitude(distance_to_center),
+                              squared_function_of_time, root_factor);
```

Comment:
I think this part can be simplified down a bit more to (untested)

```cpp
double root_factor = 0.0;
for (size_t i=0; i<Dim; i++) {
  root_factor += -2.0 * gsl::at(result, i) * function_of_time[i];
}
const double root = root_finder(magnitude(result - center),
                                sqrNorm(function_of_time),
                                root_factor));
```

In general, but especially in the maps because they are called very often, we want to avoid doing unnecessary allocations and copies as this can negatively impact performance.